idx/idx.go

```diff
@@ -70,8 +83,12 @@ type MetricIndex interface {
 	// If the pattern matches a branch node, then
 	// all leaf nodes on that branch are deleted. So if the pattern is
 	// "*", all items in the index are deleted.
-	// It returns a copy of all of the Archives deleted.
-	Delete(orgId uint32, pattern string) ([]Archive, error)
+	// It returns the number of Archives that were deleted.
+	Delete(orgId uint32, pattern string) (int, error)
+
+	// DeletePersistent deletes items from the index
+	// It returns the deleted items in a []idx.Archive
+	DeletePersistent(orgId uint32, pattern string) ([]Archive, error)
```

Comment:
do we need this? MetricIndex is a pretty core interface. changes to it shouldn't be taken lightly

Reply:
We do need this. We needed to add DeletePersistent to separate it from the regular Delete in order to return an int instead of a [] of deleted Archives/Definitions. Due to the addition of the PartitionedIndex I had to add this in order to still access it from cassandra and bigtable stores. The way they access the memory index changed.